lib/std/meta.zig

```diff
@@ -213,6 +213,124 @@ fn testSentinel() void {
     testing.expect(sentinel(*const [5]u8) == null);
 }
 
+
+/// Return the child type of the given type.  This differs from `Child` in that it handles "One Pointer"
+/// and Optional types by returning the grandchild type instead of the direct child type.
+pub fn ArrayChild(comptime T: type) type {
```

Comment:
This is very similar to what `Elem` does. Your version won't `@compileError` on `*?u8`.
I am not convinced accepting optional types is always desirable.
BTW, `Elem` could use some examples in its doc comment.

Reply:
Yes it will error on `*?u8`, because it will call `Child` on `u8` and you'll get an error.  Supporting optional types is actually quite common because nullable pointers are represented as optionals.  Take a look at the test cases.

Reply:
Thanks for finding this problem, I believe I fixed it.

Reply:
Oh my goodness I think I made another mistake reading your comment.  I was reading "Elm" as in "The Elm Programming language", not "Elem" the function!  I think you're right that this `Elem` function was already what I needed, I'll try using that.

Reply:
Ok I've modified it to only accept optional's if they are many pointer types (i.e. `?[*]u8`).   If any case makes sense to support I think it's this one.  The rest I am also not convinced are desirable to support.